tahowallet/taho.xyz
# 🤖🐬Uninstall analytics event on goodbye page visit

## What 

This PR installs the posthog js library with a [gatsby plugin](https://www.gatsbyjs.com/plugins/gatsby-plugin-posthog/)
This will start tracking website user events. 

The goodbye page accepts a `uuid` query param, and will send an `Uninstall` event with the `uuid` as `distinct_id`. 
This url will be called when the extension is uninstalled. 

## How 

- most of the auto event tracking are turned off [4c59201](https://github.com/tallycash/tallyho.org/pull/124/commits/4c592011363a3f738f9b036a30c86b7e884df75b) so we don't start collecting info automatically that we shouldn't 
- persistance is set to `localStorage` to avoid accidental data leaks through cookies

## Test

- website 
  - Open https://deploy-preview-124--tally-cash.netlify.app/goodbye?uuid=some_random_string
  - Watch the `Uninstall` event to arrive to posthog and the Person should be the `some_random_string`
  - You should not see extra events from the webpage
- website + extension
  - checkout #2676
  - set the env `SUPPORT_ANALYTICS`, `ENABLE_ANALYTICS_DEFAULT_ON`, `USE_ANALYTICS_SOURCE` env variables
  - set the `WEBSITE_ORIGIN` to `https://deploy-preview-124--tally-cash.netlify.app`
  - build > install > uninstall
  - the goodbye page should show
  - check the uuid on the `/goodbye` page url
  - check that you see `Uninstall` event in `posthog` with the given uuid
